### ISSUE TITLE
Object Cache "Get" Method incorrectly sets cache value

### ISSUE THREAD

**Issue by charlesLF** [szepeviktor/fix-w3tc#438]
I came over this issue from the official W3TC support forums: https://wordpress.org/support/topic/bug-in-w3-object-cache/

To reproduce:
```php
<?php
$test_key = “test_” . uniqid();

wp_cache_delete( $test_key, “test” );

$found = NULL;

$val = wp_cache_get( $test_key, “test”, false, $found );

var_dump($val);
var_dump($found);
```

Expected output:
```
bool(false)
bool(false)
```

Actual outcome:
```
bool(false)
bool(true)
```

The reason behind this is that:
1. function `delete` in `ObjectCache_WpObjectCache_Regular.php` starts with:
```php
if ( !$force && $this->get( $id, $group ) === false ) {
    return false;
}
```
2. Function `get` in that same file then contains the following line:
```php
$this->cache[$key] = $value;
$this->cache_total++;

if ( $value !== false ) {
    $this->cache_hits++;
} else {
    $this->cache_misses++;
}
```
3. This is setting the value for the `$key` even if `$value` is false.

My proposed fix is to only set the cache value for `$key` if `$value` is not `false` 

```php
$this->cache_total++;

if ( $value !== false ) {
    $this->cache[$key] = $value;
    $this->cache_hits++;
} else {
    $this->cache_misses++;
}
```

**Comment by nigrosimone** [comment]
Thanks!  You are right!

**Comment by nigrosimone** [comment]
merged https://github.com/szepeviktor/w3-total-cache-fixed/pull/439

thanks!
